feat: add C implementation for `stats/base/dists/levy/pdf`

NishchayRajput opened the pull request:
Resolves #3729.

## Description

> What is the purpose of this pull request?

This pull request:

-   Add C implementation for `@stdlib/stats/base/dists/levy/pdf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3729

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

NishchayRajput committed:
feat: add C implementation for @stdlib/stats/base/dists/levy/pdf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
Merge branch 'develop' into nishchay-base-levy-pdf

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
fix: address review issues in `stats/base/dists/levy/pdf` C implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
fix: update native benchmark to directly generate values in desired range

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/README.md:
```diff
@@ -136,6 +136,113 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_levy_pdf( x, mu, c )
+
+Returns the [probability density][pdf] for a [Lévy][levy-distribution] distribution with input value `x`, location parameter `mu` and scale parameter `c`.
```
```suggestion
Evaluates the [probability density][pdf] for a [Lévy][levy-distribution] distribution with input value `x`, location parameter `mu`, and scale parameter `c`.
```

Planeshifter committed:
fix: update README and C example to use direct uniform generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,74 @@
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		mu[ i ] = uniform( -10.0, 10.0 );
+		x[ i ] = uniform( mu[ i ], mu[ i ] + 40.0 );
```
```suggestion
		x[ i ] = uniform( mu[ i ], 40.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,74 @@
+	for ( i = 0; i < len; i++ ) {
+		mu[ i ] = uniform( -10.0, 10.0 );
+		x[ i ] = uniform( mu[ i ], mu[ i ] + 40.0 );
+		c[ i ] = uniform( EPS, 5.0 + EPS );
```
```suggestion
		c[ i ] = uniform( EPS, 5.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/README.md:
```diff
@@ -124,9 +124,9 @@ var y;
-    c = ( randu()*10.0 ) + EPS;
+    mu = uniform( 0.0, 10.0 );
+    x = uniform( mu, mu + 10.0 );
+    c = uniform( EPS, 10.0 + EPS );
```
```suggestion
    c = uniform( EPS, 10.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/README.md:
```diff
@@ -136,6 +136,113 @@ for ( i = 0; i < 10; i++ ) {
+    for ( i = 0; i < 25; i++ ) {
+        mu = random_uniform( 0.0, 10.0 );
+        x = random_uniform( mu, mu + 10.0 );
+        c = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 + STDLIB_CONSTANT_FLOAT64_EPS );
```
```suggestion
        c = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/examples/c/example.c:
```diff
@@ -0,0 +1,43 @@
+	for ( i = 0; i < 25; i++ ) {
+		mu = random_uniform( 0.0, 10.0 );
+		x = random_uniform( mu, mu + 10.0 );
+		c = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 + STDLIB_CONSTANT_FLOAT64_EPS );
```
```suggestion
		c = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 );
```

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
fix: update C benchmark to use direct uniform generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/pdf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	for ( i = 0; i < 100; i++ ) {
+		mu[ i ] = random_uniform( -10.0, 10.0 );
+		x[ i ] = random_uniform( mu[ i ], mu[ i ] + 40.0 );
+		c[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 5.0 + STDLIB_CONSTANT_FLOAT64_EPS );
```
```suggestion
		c[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 5.0 );
```

Planeshifter committed:
bench: simplify upper value

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thanks, @NishchayRajput!

Planeshifter commented:
/stdlib merge
